sdc_client/base_api_client.py

```diff
@@ -0,0 +1,131 @@
+from abc import abstractmethod
+
+import json
+import os
+import time
+import urllib.parse
+import inject
+import requests
+from sdc_client.interfaces import ILogger, IStreamSets
+
+PREVIEW_TIMEOUT = os.environ.get('STREAMSETS_PREVIEW_TIMEOUT', 30000)
+
+
+class _BaseStreamSetsApiClient:
```

Comment:
Because of `_get_session` raises NotImplemented, I think we should mark it as @abstractmethod and inherit it class from abc.ABC. This will prevent instantiation